### PR TITLE
add x509 cert form field to saml sso edit

### PR DESCRIPTION
**Pull request by nhsiehgit**
Simple naive solution.
Just adds a new form field on SAML forms and prefills with the x509 cert

https://github.com/getsentry/sentry/assets/6186377/623715c4-e7e7-4a51-8cc2-857589eb049e

### CODE REVIEW

**Comment by nhsiehgit** [comment]
resolves https://github.com/getsentry/sentry/issues/50792

**Review by JoshFerge — COMMENTED** [review]
would it be possible to add a test or two for this?

**Comment by nhsiehgit** [comment]
test added!

File: tests/sentry/web/frontend/test_organization_auth_settings.py [hunk 1]
```diff
@@ -606,7 +611,84 @@ def test_edit_sso_settings__scim(self):
...
+
+    def setUp(self):
+        super().setUp()
+        self.user = self.create_user("foobar@sentry.io")
+        # self.organization = self.create_organization(owner=self.user, name="saml2-org")
+        # self.auth_provider_inst = AuthProvider.objects.create(
+        #     provider=self.provider_name,
+        #     config=dummy_provider_config,
+        #     organization_id=self.organization.id,
+        # )
```
Comment (JoshFerge) [inline tests/sentry/web/frontend/test_organization_auth_settings.py:627]:
```suggestion
```

**Review by saponifi3d — APPROVED** [review]
lgtm

File: src/sentry/web/frontend/organization_auth_settings.py [hunk 2]
```diff
@@ -152,7 +167,23 @@ def handle_existing_provider(
...
+                                    **original_idp,
+                                    "x509cert": value,
+                                },
+                            }
+                            auth_service.update_provider_config(
+                                organization_id=organization.id,
+                                auth_provider_id=auth_provider.id,
+                                config=auth_provider.config,
+                            )
+                            changed_data["x509cert"] = f"to {value}"
```
Comment (JoshFerge) [inline src/sentry/web/frontend/organization_auth_settings.py:185]: should this just be above and not in this loop? i see that's how the other values that get change end up getting used. (e.g. we just always call change_scim / update_default_role)

File: tests/sentry/web/frontend/test_organization_auth_settings.py [hunk 3]
```diff
@@ -606,7 +611,84 @@ def test_edit_sso_settings__scim(self):
...
+        return organization, auth_provider
+
+    def create_om_and_link_sso(self, organization):
+        with assume_test_silo_mode(SiloMode.REGION):
+            om = OrganizationMember.objects.get(user_id=self.user.id, organization=organization)
+            setattr(om.flags, "sso:linked", True)
+            om.save()
+        return om
+
+    def test_edit_sso_settings(self):
```
Comment (JoshFerge) [inline tests/sentry/web/frontend/test_organization_auth_settings.py:647]: should we be checking that the x509 on the backend has changed? right now it seems like we're only looking at the audit log entry

File: src/sentry/web/frontend/organization_auth_settings.py (hunk 2 above)
Comment (nhsiehgit) [reply src/sentry/web/frontend/organization_auth_settings.py:185]: Yea, although we'd still want to catch the audit log change, so if we're iterating through here anyways 🤷

**Comment by nhsiehgit** [comment]
@JoshFerge cool,
addressed comments!

**Review by JoshFerge — APPROVED** [review]
thanks for the updates! lgtm

**Pull request merged**